pkg/streamingpromql/testdata/ours-only/functions.test

```diff
@@ -5,3 +5,15 @@ load 6m
 # Currently prometheus does not merge series: https://github.com/prometheus/prometheus/issues/15114
 eval range from 0 to 6m step 6m label_replace(series, "idx", "replaced", "idx", ".*")
   series{label="a", idx="replaced"} 2 4
+
+clear
+
+# label_join() tests
+load 5m
+  dup{label="a", this="a"} 1.0
+  dup{label="b", this="a"} 1.0
+
+# Prometheus fails this with enableDelayedNameRemoval disabled
+# Can be tested against both once https://github.com/prometheus/prometheus/pull/15975 is merged
```

Comment:
Looks like this has been merged - would you be able to vendor the fix into Mimir so we don't need to come back and move this later?

Reply:
I think we can keep this as-is for now, turns out there are a bunch of other changes that need to be vendored in first 😞